File: Data Structures & Algorithms/partition-equal-subset-sum/submission-0.py
class Solution:
    def canPartition(self, nums: List[int]) -> bool:
        s = sum(nums)
        if s%2==1:
            return False
        target  = s//2
        dp = [False]*(target+1)
        dp[0] = True
        for x in nums:
            for s in range(target,x-1,-1):
                dp[s] = dp[s] or dp[s-x]
            if dp[target]:
                return True
        return dp[target]

File: Data Structures & Algorithms/minimum-size-subarray-sum/submission-0.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int l = 0;
        int total = 0;
        int res  = Integer.MAX_VALUE;
        for(int right = 0; right<nums.length;right++){

            total+=nums[right];
            while(total>=target){ // because we need to keep incrementing left ptr till we meet condition
                res = Math.min(right-l+1,res);
                total-=nums[l];
                l++;
            }

        }
        if(res==Integer.MAX_VALUE)return 0;
        else return res;
    }
}